build-system/tasks/check-links.js

```diff
@@ -134,10 +118,6 @@ function checkLinksInFile(file) {
       }
       let containsDeadLinks = false;
       for (const {link, status, statusCode} of results) {
-        // Skip links to files that were introduced by the PR.
-        if (isLinkToFileIntroducedByPR(link)) {
-          continue;
```

Comment:
Instead of doing a `continue` here, maybe you could remove the `https://github.com/ampproject/amphtml/master/` prefix from the link (if it exists), normalize the rest of the path, and see if the link resolves correctly?